Mathlib/Topology/MetricSpace/Katetov.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2024 Luigi Massacci. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Luigi Massacci
+-/
+
+import Mathlib.Topology.MetricSpace.Basic
+import Mathlib.Topology.UniformSpace.Cauchy
+
+/-!
+# Katetov Maps
+
+In this file we define Katetov maps (i.e. one point extensions of a metric) and establish
+basic properties of their space. We embed a metric space in its space of Katetov maps via
+the Kuratowski embedding.
+
+## Notation
+
+ - `E(X)` is the type of Katetov maps on `X`.
+
+## References
+
+* [melleray_urysohn_2008]
+-/
+
+variable {α : Type _} [MetricSpace α]
+
+/-- A real valued function from a metric space is katetov if
+  it satisfies the following inequalities: -/
```

Comment:
```suggestion
/-- A real valued function `f` on a metric space is Katetov if
  it is 1-Lipschitz and for all `x` and `y` we have `dist x y ≤ f x + f y`. -/
```
The docstrings below will not show up if someone hovers over `IsKatetov` in another file. That's why it's good to include the properties in this docstring as well.